ambry-commons/src/main/java/com.github.ambry.commons/Account.java

```diff
@@ -0,0 +1,145 @@
+   * @param name The name of the account, which is the external reference to the account.
+   * @param metadata The metadata of the account in JSON. Can be {@code null}.
+   */
+  public Account(short id, String name, JSONObject metadata) {
+    if (name == null) {
+      throw new IllegalArgumentException("accountName cannot be null");
+    }
+    this.id = id;
+    this.name = name;
+    this.metadata = metadata;
+    containerIdToContainerMap = new HashMap<>();
+    containerNameToContainerMap = new HashMap<>();
+  }
+
+  /**
+   * Gets the id of the account.
+   * @return The id of the account.
+   */
+  public short id() {
+    return id;
+  }
+
+  /**
+   * Gets the name of the account.
+   * @return The name of the account.
+   */
+  public String name() {
+    return name;
+  }
+
+  /**
+   * Gets the metadata of the account in JSON format.
+   * @return The metadata of the account in JSON format.
+   */
+  public JSONObject metadata() {
+    return metadata;
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container id.
+   * @param containerId The id of the container to get.
+   * @return The {@link Container} of this account with the specified id, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerByContainerId(short containerId) {
+    return containerIdToContainerMap.get(containerId);
+  }
+
+  /**
+   * Gets the {@link Container} of this account with the specified container name.
+   * @param containerName The name of the container to get.
+   * @return The {@link Container} of this account with the specified name, or {@code null} if such a
+   *                    container does not exist.
+   */
+  public Container getContainerByContainerName(String containerName) {
+    return containerNameToContainerMap.get(containerName);
+  }
+
+  /**
+   * Gets all the containers of this account in a list.
+   * @return All the containers of this account.
+   */
+  public List<Container> getAllContainers() {
+    return Collections.unmodifiableList(new ArrayList<>(containerIdToContainerMap.values()));
+  }
+
+  /**
+   * Adds a container to this account.
+   * @param container The container to this account.
+   */
+  void addContainer(Container container) {
+    containerIdToContainerMap.put(container.id(), container);
+    containerNameToContainerMap.put(container.name(), container);
+  }
+
+  /**
+   * Generates a {@link String} representation that uniquely identifies this account. The string
+   * is in the format of {@code Account[id:name]}.
+   * @return The {@link String} representation of this account.
+   */
+  @Override
+  public String toString() {
+    return "Account[" + id() + ":" + name() + "]";
+  }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+
```

Comment:
@nsivabalan not sure whats to fix since the formatter is ok with this too and the code is auto generated (assuming the formatter has been run)